### PR TITLE
feat(cardeditor): add maximumDataPoints to data series cards

### PR DESCRIPTION
**Pull request by scottdickerson**
Closes #

image

**Summary**

- Support the ability to capture the maximumDataPoints a DataSeriesCard should show.  This is important because the TimeSeriesCards can reach scale issues once we get to a certain size

**Change List (commits, features, bugs, etc)**

- feat(DataSeriesFormSettings): add the property to capture maximumDataPoints for the card
- feat(DashboardEditor): support passing the 18n to upstream default cards 
- feat(index): export the getDefaultCard utility function so downstream consumers can use it in addition to their own getDefaultCard callbacks in the DashboardEditor

**Acceptance Test (how to verify the PR)**

- Verify that a TimeSeriesCard in the card editor has a new setting to capture the maximumDataPoints

**Regression Test (how to make sure this PR doesn't break old functionality)**

- Verify the existing getDefaultCard callbacks still work

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormSettings.jsx [hunk 1]
```diff
@@ -30,6 +30,8 @@ const propTypes = {
       includeZeroOnYaxis: PropTypes.bool,
       timeDataSourceId: PropTypes.string,
       showLegend: PropTypes.bool,
+      /** maximum amounts of data points to render in the graph */
+      maximumDataPoints: PropTypes.bool,
```
Comment (kevinsperrine) [inline packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormSettings.jsx:34]: bool?

**Review by kevinsperrine — COMMENTED** [review]
Code looks good. Is `bool` the correct prop type? Just for my own clarity, how does this actually get applied to the card? For example, if I edit the [dashboard editor with initial value story](https://deploy-preview-3116--carbon-addons-iot-react.netlify.app/?path=/story/2-watson-iot-experimental-%E2%98%A2%EF%B8%8F-dashboardeditor--with-initial-value), and set a maximum number of data points of 3 on the TimeSeriesCard. I still see 14 data points in the chart, and I'm able to add Torque, firmware, and manufacturer as additional data sets, so the chart grows to 35 data points across all five types.

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormSettings.jsx (hunk 1 above)
Comment (scottdickerson) [reply packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormSettings.jsx:34]: nope, thanks for catching.  The data point limitation doesn't actually affect the cards themselves currently, only is stored in the JSON and up to the downstream consumers to use when they fetch their data.  Should I add to the chart itself?

Comment (kevinsperrine) [reply packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormSettings.jsx:34]: Got it. No, I don't think it's necessary to add to the chart itself, but maybe clarify that in the prop-type comment?

**Review by bjornalm — APPROVED** [review]
Looks good once Kevin's comments have been addressed

**Review by davidicus — APPROVED** [review]

**Pull request merged**